tests/test_exacttarget_field_selection.py

```diff
@@ -0,0 +1,134 @@
+            "list_send": ["MissingAddresses", "ExistingUndeliverables", "HardBounces", "NumberDelivered"],
+            "list_subscriber": ["Status", "ObjectID", "ListID"],
+            "send": ["Status", "EmailName", "FromAddress", "IsMultipart"],
+            "subscriber": ["Status", "EmailAddress", "SubscriberKey", "PartnerKey"]
+        }
+
+    def name(self):
+        return "tap_tester_exacttarget_field_selection"
+
+    def test_run(self):
+        # run test with all fields of stream 'selected = False'
+        self.run_test(only_automatic_fields=True)
+        # run test with fields in 'non_selected_fields', 'selected = False'
+        self.run_test(only_automatic_fields=False)
+
+    def run_test(self, only_automatic_fields=False):
+        expected_streams = self.streams_to_select()
+        conn_id = connections.ensure_connection(self)
+        runner.run_check_mode(self, conn_id)
+
+        expected_stream_fields = dict()
+
+        found_catalogs = menagerie.get_catalogs(conn_id)
+        for catalog in found_catalogs:
+            stream_name = catalog['stream_name']
+            catalog_entry = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+            if not stream_name in expected_streams:
+                continue
+            # select catalog fields
+            self.select_found_catalogs(
+                conn_id,
+                [catalog],
+                only_streams=[stream_name],
+                deselect_all_fields=True if only_automatic_fields else False,
+                non_selected_props=[] if only_automatic_fields else self.non_selected_fields[stream_name])
+            # add expected fields for assertion
+            fields_from_field_level_md = [md_entry['breadcrumb'][1]
+                                          for md_entry in catalog_entry['metadata']
+                                          if md_entry['breadcrumb'] != []]
+            if only_automatic_fields:
+                expected_stream_fields[stream_name] = self.expected_primary_keys()[stream_name] | self.expected_replication_keys()[stream_name]
+            else:
+                expected_stream_fields[stream_name] = set(fields_from_field_level_md) - set(self.non_selected_fields[stream_name])
+
+        self.run_and_verify_sync(conn_id)
+        synced_records = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # get expected keys
+                expected_keys = expected_stream_fields[stream]
+
+                # collect all actual values
+                messages = synced_records.get(stream)
+
+                # collect actual synced fields
+                actual_keys = [set(message['data'].keys()) for message in messages['messages']
+                                   if message['action'] == 'upsert'][0]
+
+                if stream == 'list':
+                    expected_keys = expected_keys - {
```

Comment:
@harshpatel4crest Why are we extracting fields like 'SendClassification', 'PartnerProperties', etc. from the expected kets?

Reply:
Yes, they are not retrievable. So, added a comment regarding that.